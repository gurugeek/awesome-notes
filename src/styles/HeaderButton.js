export default {
    wrapper: {
        marginHorizontal: 10
    }
};