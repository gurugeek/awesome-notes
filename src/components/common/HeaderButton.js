import React from 'react';
import { TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import PropTypes from 'prop-types';

import { ICON_TYPES } from '../../constants/IconTypes';
import { HeaderButtonStyle } from '../../styles';

const HeaderButton = ({ action, type }) => {
    return (
        <TouchableOpacity onPress={action} style={HeaderButtonStyle.wrapper}>
            <Icon name={ICON_TYPES[type].name} size={40} color={ICON_TYPES[type].color} />
        </TouchableOpacity>
    );
};

HeaderButton.propTypes = {
    action: PropTypes.func,
    type: PropTypes.string.isRequired
};

export default HeaderButton;