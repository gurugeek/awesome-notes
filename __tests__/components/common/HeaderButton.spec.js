import React from 'react';
import { TouchableOpacity } from 'react-native';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Icon from 'react-native-vector-icons/Ionicons';

Enzyme.configure({ adapter: new Adapter() });

import HeaderButton from '../../../src/components/common/HeaderButton';

describe('<HeaderButton /> Component', () => {
    it('HeaderButton should match with snapshot', () => {
        const tree = renderer.create(
            <HeaderButton type='plus' action={() => {}} />
        );
        const json = tree.toJSON();
        expect(json).toMatchSnapshot();
    });

    const headerButtonShallow = shallow(<HeaderButton type='plus' action={() => {}} />);

    it('HeaderButton should exists', () => {
        expect(headerButtonShallow.length).toEqual(1);
    });

    it('HeaderButton should has one TouchableOpacity', () => {
        expect(headerButtonShallow.find(TouchableOpacity)).toHaveLength(1);
    });

    it('HeaderButton should has one Icon', () => {
        expect(headerButtonShallow.find(Icon)).toHaveLength(1);
    });
});