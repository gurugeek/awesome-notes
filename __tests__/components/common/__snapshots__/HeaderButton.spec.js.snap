// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderButton /> Component HeaderButton should match with snapshot 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginHorizontal": 10,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 40,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    ï„‚
  </Text>
</View>
`;
